### PR TITLE
RHMAP-4978 - Making JS SDK not init automatically and use fh.init instead

### PR DESCRIPTION
**Pull request by jcesarmobile**
I've been trying different approaches to make to init the SDK manually and I think this one is the better way of doing it.

I'm reusing the existing init on feedhenry.js as it was going to be deprecated (talked with @wei-lee about it), but removed the first param (opts), so it only have two params, the success callback and the fail callback (as the other SDKs)
In case it fails, it returns the error
In case it succeed it doesn't return anything (as the other SDKs don't return anything neither)

I've moved the fhinit function on src/modules/waitForCloud.js from inside the ready function and made it available on the exports, and I call it on the feedhenry.js init.

I've added an error on src/modules/waitForCloud.js ready method to be called when you try to use the cloud, but cloud is not ready and is not being initialized at the moment (i.e. if you try to execute cloud functions before manually calling init).

### CODE REVIEW

File: src/feedhenry.js [hunk 1]
```diff
@@ -46,17 +46,21 @@ var once = function(type, listener) {
...
+var init = function(success, fail) {
+  if(typeof window.cordova !== "undefined" || typeof window.phonegap !== "undefined"){
+    //if we are running inside cordova/phonegap, only init after device is ready to ensure the device id is the right one
+    document.addEventListener("deviceready", cloud.fhinit, false);
+  } else {
+    cloud.fhinit();
+  }
   cloud.ready(function(err, host) {
     if (err) {
       if (typeof fail === "function") {
```
Comment (pb82) [inline src/feedhenry.js]: No need for nested ifs here, could also be written as `if (err && typeof fail === "function")`

**Comment by pb82** [comment]
LGTM, although i don't have much knowledge about the JS SDK.

**Comment by corinnekrych** [comment]
@jcesarmobile could you give an example of how template will use it? have you tried it on template apps?

**Comment by jcesarmobile** [comment]
I've tried it on the Cordova Hello World Project

The simple example: `$fh.init(function(){alert('init succeeded');},function(){alert('init failed');});`

**Comment by corinnekrych** [comment]
LGTM

**Comment by edewit** [comment]
👀

**Comment by edewit** [comment]
👍

**Comment by aidenkeating** [comment]
@jcesarmobile Is this change still required or can it be closed along with its JIRA task? https://issues.jboss.org/browse/RHMAP-4978

**Comment by jcesarmobile** [comment]
It's required

**Comment by wtrocki** [comment]
@jcesarmobile - I was thinking about doing the same then I saw that PR. 
Really good job on this. This change will be really useful and also can resolve multiple problems end users are facing.  Any idea why this wasn't merged?

**Comment by jcesarmobile** [comment]
Sorry @wtrocki, I missed your comment.

I sent an email about it and we decided to focus on other problems with more priority and also we didn't want to introduce breaking changes by then. A change like this will require to update all the templates and might take a lot of time that we didn't have by that time.
I sent it by 3.17 time frame, maybe we should start planning this to be available in 3.19 or 3.20.
Can you answer on the mail I sent?

**Comment by wtrocki** [comment]
Cannot find that email. 

In my opinion having sdk automatically mounting some routes, configurations etc. is anti pattern. 
This change will be the one step forward for more cleaner SDK where developers decide what they can use and when. I fully understand that this change will include a lots of testing and updates. Just wanted to point that this is the way to go.

**Comment by camilamacedo86** [comment]
@corinnekrych @wtrocki @pb82 the people which worked in this PR so long are with us. Should we keep this one and move forward or should we close this PR because it is out of date? Could you give a hand here?

**Comment by wtrocki** [comment]
Good to close from my point of view. 
We can reopen if needed.

**Pull request closed without merging**